blog: How to Filter, Map, Sort, and Reduce Data in Node-RED (#2203)
FlowFuse/website: src/blog/2024/06/filtering-mapping-sorting-reducing-with-node-red.md

---
title: How to Filter, Map, Sort, and Reduce Data in Node-RED
subtitle: Step-by-step guide to use the most common data transformation operations like filtering, mapping, sorting and reducing in Node-RED.
description: Learn how to perform data transformation in Node-RED with a low-code approach.

> date: 2024-06-19

Reviewer: ```suggestion
date: 2024-06-21
```

---
title: How to Filter, Map, Sort, and Reduce Data in Node-RED

> subtitle: Step-by-step guide on how to perform filtering, mapping, sorting and reducing in Node-RED.

Reviewer: ```suggestion
subtitle: Step-by-step guide to use the most common data transformation operations like filtering, mapping, sorting and reducing in Node-RED.
```

description: Learn how to perform data transformation in Node-RED with a low-code approach.
date: 2024-06-19
authors: ["sumit-shinde"]
image: 
tags:
   - posts
   - node-red
   - flowfuse
   - low-code
   - filtering
   - sorting
   - mapping 
   - reducing
   - data tranformation
---
   

> [Node-RED](/node-red/) is a visual programming platform known for its user-friendly interface and low-code approach, that allows developers to spend more time developing solutions instead of dealing with technical complexities. However, users can sometimes make applications developed with Node-RED complex and hard to maintain when they are not familiar with its basics. Understanding Node-RED basics helps keep flows low-coded.

Reviewer: I think it's not about developers, it's about expanding who can build software.

<!--more-->

Often, I have noticed that many users are unfamiliar with how to filter, map, sort, and reduce data using a low-code approach. They tend to rely on function nodes, which can add complexity. If you are unfamiliar with function nodes and their drawbacks, please refer to [The benefits and drawbacks of using Node-RED function nodes](/blog/2023/03/why-should-you-use-node-red-function-nodes/). In this guide, we will show you how to filter, map, sort, and reduce data with Node-RED while sticking to low-code practices.

## What is Low-Code

Low-code is a software development approach that requires little to no coding to build applications and processes. Instead of using complex programming languages, you use visual interfaces with basic logic and drag-and-drop capabilities.

> Low-code is not just about accelerating development; it’s about democratizing it. It’s about giving more people the ability to create solutions to business problems.
>
> *— Charles Lamanna, Corporate Vice President of Business Applications & Platforms at Microsoft*

> For more details refer to the following guides:

Reviewer: They don't guide people, these links :)
```suggestion
For more details refer to the following articles:
```

- [Why Low-Code is Better](/blog/2024/03/low-code-is-better/).
- [Why you need a low-code platform](/blog/2024/05/why-you-need-a-low-code-platform/).

## Why do you need to learn to filter, map, sort, and reduce the data?

Learning to filter, map, sort, and reduce data is crucial for effective data processing and transformation. For example, consider the scenario where you have an array of sensor data retrieved from an InfluxDB database. The data looks something like this:

```json

> [

Reviewer: Yes, you could map from `emperature` to `temperature`, but maybe better to map from Kelvin to Celsius, or so?

Reviewer: @sumitshinde-84 Please update this

{% endrenderFlow %}

While it may seem simple to you, it will require sufficient JavaScript knowledge for all other members involved in your project. Most Node-RED users are creative innovators who prefer developing quick and easy solutions and do not want to spend time learning programming languages.

> ## Using Low-Code No-Code approach
> 
> Now let's do it with the standard Node-RED low-code approach:

Reviewer: ```suggestion
```

   - sorting
   - mapping 
   - reducing
   - data transformation
---
   

> [Node-RED](/node-red/) is a visual programming platform known for its user-friendly interface and low-code approach, that allows developers to spend more time developing solutions instead of dealing with technical complexities. However, users can sometimes make applications developed with Node-RED complex and hard to maintain when they are not familiar with its basics. Understanding Node-RED basics helps keep flows low-coded which helps keep your projects simple and easy to understand. This enables more team members to contribute ideas and develop effective solutions for business problems.

Author: Because most of the time, not knowing the NR LC way of performing these operations leads to using the function node. That's why adding information about LC is important.

> [Node-RED](/node-red/) is a visual programming platform known for its user-friendly interface and low-code approach, that allows developers to spend more time developing solutions instead of dealing with technical complexities. However, users can sometimes make applications developed with Node-RED complex and hard to maintain when they are not familiar with its basics. Understanding Node-RED basics helps keep flows low-coded which helps keep your projects simple and easy to understand. This enables more team members to contribute ideas and develop effective solutions for business problems.

Author: Some users also believe that using the function node is an LC approach.

> [Node-RED](/node-red/) is a visual programming platform known for its user-friendly interface and low-code approach, that allows developers to spend more time developing solutions instead of dealing with technical complexities. However, users can sometimes make applications developed with Node-RED complex and hard to maintain when they are not familiar with its basics. Understanding Node-RED basics helps keep flows low-coded which helps keep your projects simple and easy to understand. This enables more team members to contribute ideas and develop effective solutions for business problems.

Reviewer: @sumitshinde-84 I agree with Hasmin, please explain why you've chosen these 3 functions. Steer away from generic, uninspired content. Why are these three so important?
